sdk/keyvault/Azure.Security.KeyVault.Certificates/tests/CertificatesTestBase.cs

```diff
@@ -15,6 +15,7 @@ public class CertificatesTestBase : RecordedTestBase
     {
         public const string AzureKeyVaultUrlEnvironmentVariable = "AZURE_KEYVAULT_URL";
         private readonly HashSet<string> _toCleanup = new HashSet<string>();
+        private readonly ReaderWriterLockSlim _lock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
```

Comment:
nit: ReaderWriterLockSlim seems a bit heavy handed here. why not just use a lock statement.

Reply:
If/when we run these in parallel, we don't want to lock the thread more than necessary. Here I'm taking advantage of the read and separate writer locks.